src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,88 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import classNames from 'classnames';
+import { actions as userActions } from '../../features/user';
+
+interface Props {
+  currentTodo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({ currentTodo }) => {
+  const { id, title, completed, userId } = currentTodo;
+  const dispatch = useAppDispatch();
+  const { user, isLoading } = useAppSelector(state => state.user);
+
+  useEffect(() => {
+    dispatch(userActions.setLoading(true));
+    getUser(userId)
+      .then(userFromServer => dispatch(userActions.setUser(userFromServer)))
+      .catch(() => {
+        dispatch(userActions.setError('Error'));
```

Comment:
The error handling in the `catch` block uses a hardcoded error message ('Error'). Consider providing more details about the error or logging the error object to improve debugging capabilities.